microsoft/testsuites/vm_extensions/azurediskencryption.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import json
+import time
+
+from assertpy import assert_that
+from azure.mgmt.keyvault.models import AccessPolicyEntry, Permissions
+from azure.mgmt.keyvault.models import Sku as KeyVaultSku  # type: ignore
+from azure.mgmt.keyvault.models import VaultProperties
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.operating_system import SLES, CBLMariner, CentOs, Oracle, Redhat, Suse, Ubuntu
+from lisa.sut_orchestrator import AZURE
+from lisa.sut_orchestrator.azure.common import (
+    AzureNodeSchema,
+    create_keyvault,
+    get_sp_object_id,
+    get_tenant_id,
+)
+from lisa.sut_orchestrator.azure.features import AzureExtension
+from lisa.sut_orchestrator.azure.platform_ import AzurePlatform, AzurePlatformSchema
+from lisa.testsuite import TestResult
+
+
+def _enable_ade_extension(node: Node, log: Logger, result: TestResult) -> any:
+    environment = result.environment
+    assert environment, "fail to get environment from testresult"
+    platform = environment.platform
+    assert isinstance(platform, AzurePlatform)
+    runbook = platform.runbook.get_extended_runbook(AzurePlatformSchema)
+    tenant_id = get_tenant_id(platform.credential)
+    if tenant_id is None:
+        raise ValueError("Environment variable 'tenant_id' is not set.")
+    object_id = get_sp_object_id(runbook.service_principal_client_id)
```

Comment:
The runbook.service_principal_client_id should be the same as object_id. And the runbook.service_principal_client_id is optional, what happened if it's not set?